opal-ui/src/components/admin/providers/AddIdentityProviderDialog.vue

```diff
@@ -0,0 +1,348 @@
+          >
+          </q-input>
+          <q-input
+            v-model="newProvider.secret"
+            dense
+            type="text"
+            :label="$t('identity_provider.secret') + '*'"
+            :hint="$t('identity_provider.secret_hint')"
+            class="q-mb-md"
+            lazy-rules
+            :rules="[validateRequiredField('validation.identity_provider.secret_required')]"
+          >
+          </q-input>
+          <q-input
+            v-model="newProvider.discoveryURI"
+            dense
+            type="text"
+            :label="$t('identity_provider.discovery_uri') + '*'"
+            class="q-mb-md"
+            lazy-rules
+            :rules="[
+              validateRequiredField('validation.identity_provider.discovery_uri_required'),
+              validateUri('validation.identity_provider.discovery_uri_format', true),
+            ]"
+          >
+            <template v-slot:hint>
+              <div v-html="$t('identity_provider.discovery_uri_hint', { url: discoveryURIDefinition })"></div>
+            </template>
+          </q-input>
+          <q-input
+            v-model="newProvider.label"
+            dense
+            type="text"
+            :label="$t('label')"
+            :hint="$t('identity_provider.label_hint')"
+            class="q-mb-md"
+            lazy-rules
+          >
+          </q-input>
+          <q-input
+            v-model="newProvider.providerUrl"
+            dense
+            type="text"
+            :label="$t('identity_provider.provider_url')"
+            :hint="$t('identity_provider.provider_url_hint')"
+            class="q-mb-md"
+            lazy-rules
+            :rules="[validateUri('validation.identity_provider.provider_url_format', false)]"
+          >
+          </q-input>
+          <q-input
+            v-model="newProvider.groups"
+            dense
+            type="text"
+            :label="$t('groups')"
+            :hint="$t('identity_provider.groups_hint')"
+            class="q-mb-md"
+            lazy-rules
+          >
+          </q-input>
+          <q-input
+            v-model="newProvider.scope"
+            dense
+            type="text"
+            :label="$t('identity_provider.scope')"
+            class="q-mb-md"
+            lazy-rules
+            :rules="[(val) => true]"
+          >
+            <!-- NOTE: to render hint in a with HTML the fields needs rules -->
+            <template v-slot:hint>
+              <div v-html="$t('identity_provider.scope_hint', { openid: '<code>openid</code>' })"></div>
+            </template>
+          </q-input>
+          <q-input
+            v-model="newProvider.usernameClaim"
+            dense
+            type="text"
+            :label="$t('identity_provider.username_claim')"
+            :hint="$t('identity_provider.username_claim_hint')"
+            class="q-mb-md"
+            lazy-rules
+          >
+          </q-input>
+          <q-select
```

Comment:
use emit-value and map-options attributes of q-select

Reply:
:+1: